src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,122 @@
+            }
+
+            dukeFile = new File(FILE_PATH);
+
+            if (!dukeFile.exists()) {
+                dukeFile.createNewFile();
+            }
+            Paths.get(FILE_PATH);
+        } catch (IOException e) {
+            throw new DukeException("Ugh. I cannot seem to create a file for you ...");
+        }
+    }
+
+    /**
+     * Loads the {@code duke} data from this storage file, and then returns it.
+     * Returns an empty {@code duke} if the file does not exist, or is not a regular file.
+     *
+     * @throws DukeException if there were errors reading and/or converting data from file.
+     */
+    public void load(List tasks) throws DukeException {
+        this.tasks = tasks;
+        try {
+            Scanner sc = new Scanner(dukeFile);
+            while (sc.hasNextLine()) {
+                String commands = sc.nextLine();
+                String taskType = commands.substring(1, 2);
+                String taskStatus = commands.substring(4, 5);
+                String taskDescription = commands.substring(7);
+
+                switch (taskType) {
+                    case "T":
+                        Task toDo = new ToDo(taskDescription);
+                        if (taskStatus.equals("X")) {
+                            toDo.markTaskAsDone();
+                        }
+                        tasks.addTask(toDo);
+                        break;
+                    case "D":
+                        String[] deadlineDescription = taskDescription.split("by:");
+                        String deadlineName = deadlineDescription[0]
+                                .substring(0, deadlineDescription[0].length() - 1)
+                                .trim();
+                        String by = deadlineDescription[1].substring(1, deadlineDescription[1].length() - 1).trim();
+                        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM d yyyy");
+                        LocalDate date = LocalDate.parse(by, formatter);
+                        Task deadline = new Deadline(deadlineName, date);
+                        if (taskStatus.equals("X")) {
+                            deadline.markTaskAsDone();
+                        }
+                        tasks.addTask(deadline);
+                        break;
+                    case "E":
+                        String[] eventDescription = taskDescription.split("at:");
+                        String eventName = eventDescription[0]
+                                .substring(0, eventDescription[0].length() - 1)
+                                .trim();
+                        String at = eventDescription[1].substring(0, eventDescription[1].length() - 1).trim();
+                        Task event = new Event(eventName, at);
+                        if (taskStatus.equals("X")) {
+                            event.markTaskAsDone();
+                        }
+                        tasks.addTask(event);
+                        break;
```

Comment:
Maybe can consider to have a function to parse Events and Deadline as there is a bit of code duplication here?

Reply:
Good suggestion!